### PR TITLE
188 content restriction

### PR DESCRIPTION
**Pull request by ksuhartono97**

### CODE REVIEW

File: server/api/controllers/whitelist/get-whitelist.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module.exports = {
+  friendlyName: 'Get whitelist',
+
+  description:
+    'Retrieves the list of whitelists that exist if id is not specified, finds a specific whitelist entry otherwise',
+
+  inputs: {
+    id: {
+      type: 'string',
```
Comment (albertparedandan) [inline server/api/controllers/whitelist/get-whitelist.js:9]: do we have to put `required: false` here?

File: server/config/routes.js [hunk 2]
```diff
@@ -19,70 +19,75 @@ module.exports.routes = {
...
+
+  // Whitelist APIs
+  'get /api/whitelist/get': 'whitelist/get-whitelist',
+  'post /api/whitelist/create': 'whitelist/post-create-whitelist',
+  'post /api/whitelist/update': 'whitelist/post-update-whitelist',
 
   // UI
-  "get /*": {
+  'get /*': {
     skipAssets: true,
```
Comment (albertparedandan) [inline server/config/routes.js:88]: actually, is this the reason why we cannot access assets in our server lol, like the brp stuff have to be in images folder

File: ui/src/components/admin/AdminWhitelistManager.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      <Container maxW="container.lg" minHeight="85vh" p={5}>
+        <Heading mb={6} alignItems="center">
+          Whitelist Manager
+        </Heading>
+        <Text>
+          Insert whitelist here (hardcoded for retreat, don't touch if you don't
+          know what you're doing)
+        </Text>
+        <form onSubmit={handleSubmit(submitWhitelist)}>
+          <Textarea
```
Comment (albertparedandan) [inline ui/src/components/admin/AdminWhitelistManager.js:49]: do we not want to populate this with the whitelist if it already exists in the mongo?

**Review by albertparedandan — APPROVED** [review]
no major comments, just suggestions

File: ui/src/components/admin/AdminWhitelistManager.js (hunk 3 above)
Comment (ksuhartono97) [reply ui/src/components/admin/AdminWhitelistManager.js:49]: don't think this is necessary, the interface is there to simply just quickly update if needed haha

File: server/config/routes.js (hunk 2 above)
Comment (ksuhartono97) [reply server/config/routes.js:88]: i dont think so actually lolol

**Pull request merged**